plugins/woocommerce-admin/client/customize-store/intro/services.ts

```diff
@@ -55,3 +62,43 @@ export const fetchThemeCards = async () => {
 		},
 	];
 };
+
+export const fetchIntroData = async () => {
+	const currentTemplate = await resolveSelect(
+		coreStore
+		// @ts-expect-error No types for this exist yet.
+	).__experimentalGetTemplateForLink( '/' );
+
+	const styleRevs = await resolveSelect(
+		coreStore
+		// @ts-expect-error No types for this exist yet.
+	).getCurrentThemeGlobalStylesRevisions();
+
+	const hasModifiedPages = await resolveSelect( coreStore )
+		// @ts-expect-error No types for this exist yet.
+		.getEntityRecords( 'postType', 'page', {
+			per_page: 100,
+			_fields: [ 'id', '_links.version-history' ],
+			orderby: 'menu_order',
+			order: 'asc',
+		} )
+		?.some( ( page: { _links: { [ key: string ]: string[] } } ) => {
+			return page._links?.[ 'version-history' ]?.length > 1;
+		} );
```

Comment:
It seems that it throws a TypeError

![Screenshot 2023-09-27 at 14 25 32](https://github.com/woocommerce/woocommerce/assets/4344253/a1ad06e9-acc1-444f-9325-8af129776869)

I think we need to wrap Await in parenthesis to resolve the Promise result.

```suggestion
	const hasModifiedPages = (
		await resolveSelect( coreStore )
			// @ts-expect-error No types for this exist yet.
			.getEntityRecords( 'postType', 'page', {
				per_page: 100,
				_fields: [ 'id', '_links.version-history' ],
				orderby: 'menu_order',
				order: 'asc',
			} )
	 )?.some( ( page: { _links: { [ key: string ]: string[] } } ) => {
		return page._links?.[ 'version-history' ]?.length > 1;
	} );
```

Reply:
Thank you! Updated in https://github.com/woocommerce/woocommerce/pull/40362/commits/01e2126df6cc3920c474a3f687e97d34160be3d3

(Committed before seeing this suggestion 🤦 )